### PR TITLE
Allow automatic building of DEA Tools API docs

### PR DESCRIPTION
**Pull request by robbibt**
Decause DEA Notebooks is getting more popular, it would be fantastic to have the modules and functions in the `dea_tools` Python package more visible so that users can easily work out how to use them.

Most of the hard work here has already been done, with the `dea_tools` package being well documented with consistent Numpy-formatted docstrings, e.g.: https://github.com/GeoscienceAustralia/dea-notebooks/blob/develop/Tools/dea_tools/datahandling.py#L109-L233

Recently, DEA Africa used their similar Docs page to render a nice automatically generated API guide for their DE Africa Tools directly from the function docstrings: https://docs.digitalearthafrica.org/fr/latest/sandbox/notebooks/Tools/index.html. This was created by including an `index.rst` file inside their Tools package with an autosummary section (https://github.com/digitalearthafrica/deafrica-sandbox-notebooks/blob/main/Tools/index.rst), which is then linked in their docs (https://github.com/digitalearthafrica/deafrica-docs/blob/main/sandbox/index.rst#additional-guides).
 
I have copied their approach by creating an `index.rst` file here inside `dea_tools` (containing a simple placeholder `autosummary` section), then linked to this in `dea-docs/index.rst`:
- https://github.com/GeoscienceAustralia/dea-notebooks/blob/develop/Tools/index.rst
- https://github.com/GeoscienceAustralia/dea-docs/blob/master/index.rst

To get the docs to build, I've copied the DE Africa approach of first adding `dea_tools` to the path inside `conf.py`, then mocking the dependencies of that package so that Sphinx does not fail. This works successfully locally, producing nice automatic docs:

MicrosoftTeams-image (2)
MicrosoftTeams-image (3)
MicrosoftTeams-image (4)

Other changes are Black formatting for code in `conf.py`.

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -4,50 +4,124 @@
...
 ]
 
+# Add dea_tools to Python path for autosummary doc generation
+sys.path.insert(0, os.path.abspath("./notebooks/Tools"))
+autosummary_generate = ["notebooks/Tools/index.rst"]
+autodoc_default_options = {
+    "members": True,
+}
+
+# We are are not installing dea_tools dependancies, so we need
```
Comment (jmettes) [inline conf.py:30]: dependencies

File: conf.py [hunk 2]
```diff
@@ -4,50 +4,124 @@
...
-    'sphinx.ext.doctest',
-    'sphinx.ext.intersphinx',
-    'sphinx.ext.mathjax',
-    'sphinx.ext.viewcode',
-    'sphinx.ext.todo',
-    'recommonmark',
+    "nbsphinx",
+    "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
+    "sphinx.ext.napoleon",
```
Comment (jmettes) [inline conf.py:14]: Ah, I assume this was also needed, along with the mocking and adding dea_tools path

Comment (robbibt) [reply conf.py:14]: Ah yep, I added `sphinx.ext.napolean` (as previously done by Africa) to correctly interpret the Numpy-style docstrings - without it they get formatted without the nice "Parameters" and "Returns" structure/formatting
https://github.com/digitalearthafrica/deafrica-docs/blob/main/conf.py#L35

Comment (robbibt) [reply conf.py:14]: With: 
image

Without:
image

**Review by jmettes — APPROVED** [review]

**Comment by robbibt** [comment]
Hi all, I did make one small additional change to add some extra packages to the "mock" list, because the docs were failing to build correctly for some of the other DEA Tools modules (outside of `dea_tools.datahandling`):

image

Works nicely for all of them now (will need some minor updates to docstrings to keep them consistent):

image

**Comment by robbibt** [comment]
Hi all, I'm going to merge this for now just to we can get this online in time for AEO - super happy to make any changes later on too! This is just a first pass at getting this working.

**Pull request merged**